src/bridges/zora/ZoraBridge.sol

```diff
@@ -179,8 +181,41 @@ contract ZoraBridge is BridgeBase {
             
             // Update the asset map to correlate new virtual token with existing
             // bridge-owned nft.
-            nftAssets[_interactionNonce] = token;
-            delete nftAssets[inputAssetId];
+            _updateVirtualAssetId(inputAssetId, _interactionNonce);
+
+            return (1, 0, false);
+        }
+        /* 
+            cancel: https://docs.zora.co/docs/smart-contracts/modules/Asks/zora-v3-asks-v1.1#cancelask
+          
+            _auxData = 
+                bits[0-4)   = funcSelector
+                bits[4-64)  = unused 
+        */ 
+        else if (funcSelector == 3) {
+            // Input type needs to be VIRTUAL.
+            if (_inputAssetA.assetType != AztecTypes.AztecAssetType.VIRTUAL) {
+                revert ErrorLib.InvalidInputA();
+            }
+            // Output type needs to be VIRTUAL.
+            if (_outputAssetA.assetType != AztecTypes.AztecAssetType.VIRTUAL) {
+                revert ErrorLib.InvalidOutputA();
+            }
+
+            uint256 inputAssetId = _inputAssetA.id;
```

Comment:
i think its cheaper, gas wise, just to use `_inputAssetA.id` directly in the rest of the function rather than copying it to another variable.

Reply:
discussed in https://github.com/michaelneuder/aztec-connect-bridges/pull/11#discussion_r1065263074